File: go/prob4.go
//Find the largest palindrome made from the product of two 3-digit numbers.

package main

import(
        "fmt"
        "strconv"
      )

func isPalindrome(n string) bool {
    end := len(n) - 1
    for _, x := range n {
        if int(x) == int(n[end]) {
            end--
        } else {
            return false
        }
    }
return true
}

func main () {
    palindrome := 0
    for i := 999; i >= 100; i-- {
        for n := 999; n >= 100; n-- {
            total := i * n
            if total > palindrome {
                if isPalindrome(strconv.Itoa(total)) {
                    palindrome = total
                }
            }
        }
    }
    fmt.Println(palindrome)
}

File: python/prob4.py
#!/usr/bin/env python
#Find the largest palindrome made from the product of two 3-digit numbers.

import euler

product = 0
palindrome = 0

for n in range(999,100,-1):
	for x in range(100,999,1):
		product = n * x
		if euler.is_palindrome(str(product)) == True:
			if product > palindrome:
				palindrome = product

print(palindrome)
